sql/core/src/main/scala/org/apache/spark/sql/execution/python/FlatMapGroupsInPandasExec.scala

```diff
@@ -145,7 +146,15 @@ case class FlatMapGroupsInPandasExec(
         sessionLocalTimeZone,
         pythonRunnerConf).compute(grouped, context.partitionId(), context)
 
-      columnarBatchIter.flatMap(_.rowIterator.asScala).map(UnsafeProjection.create(output, output))
+      columnarBatchIter.flatMap { batch =>
+        // Grouped Map UDF returns a StructType column in ColumnarBatch, select the children here
+        // TODO: ColumnVector getChild is protected, so use ArrowColumnVector which is public
+        val structVector = batch.column(0).asInstanceOf[ArrowColumnVector]
+        val outputVectors = output.indices.map(structVector.getChild(_).asInstanceOf[ColumnVector])
```

Comment:
There might be a better way to do this, or at least make `getChild` public in `ColumnVector` to avoid the casting here. wdyt @ueshin ?

Reply:
I think the logic itself is fine. But doesn't this mean we cannot support nested structs in grouped map Pandas UDFs?

Reply:
Another concern is tho .. I think all of Arrow implementations (including SparkR ones) dont modify the batch's outputs but use the batch as are.

Reply:
Since `ColumnVector` is a very public interface compared to `ArrowColumnVector` now, I'm not sure we should make it public.

Reply:
I think it's fine to modify the batch's outputs here, but we need to think about the previous concern from @HyukjinKwon. WDYT about that @BryanCutler ?

Reply:
>I think the logic itself is fine. But doesn't this mean we cannot support nested structs in grouped map Pandas UDFs?

Nested structs were never supported in grouped map UDFs (I verified with code prior to https://github.com/apache/spark/pull/23900).  Part of the reason for this is there is no explicit logical type for a struct in a Pandas DataFrame.  When creating a nested struct in pyarrow, then converting to pandas, the struct column gets converted to a column of dictionaries, which Spark could handle but brings some other complications. So this cleanup should keep the functionality the same.

Reply:
>Another concern is tho .. I think all of Arrow implementations (including SparkR ones) dont modify the batch's outputs but use the batch as are.

Yeah, makes the Scala side a bit different but I think it is worth it to make things in `worker.py` more consistent. With this cleanup, all of the Pandas UDFs go through the same logic to be serialized.